app/schemas/models/classroom.schema.js

```diff
@@ -94,6 +94,9 @@ _.extend(ClassroomSchema.properties, {
       { title: 'Student Lock Object', description: 'Key value of student id tied to the lock data.' }, {
         courseId: c.objectId(),
         levelOriginal: c.objectId(),
+        lockedScenarioLevels: c.object({
+          additionalProperties: ['boolean', c.stringDate()]
+        }),
```

Comment:
any reason we can't reuse existing schema by using level-ids but id will be of scenario instead 
 based on course?  Doesn't make a big difference though but want to understand the reason for separation as we will need to do something for other products going forward and I don't want too many locked keys option.

Also, a good idea to have it on course-instace itself?

Reply:
@mrfinch coderabbit commented quite a good explanation here! :) I've created the dedicated `lockedScenarioLevels` to avoid any future confusion that might happen. The object keys in `lockedLevels` will always be `level._id` values, and keys of `lockedScenarioLevels` will always be `ai_scenario._id` values.